File: tests/fixtures/generic/vulnerable/taint_ldap.java
// @audit-fixture
// @rule: taint_ldap
// @expected: detected
import javax.servlet.http.HttpServletRequest;
import javax.naming.directory.*;

public class LdapController {
    public void search(HttpServletRequest request) throws Exception {
        String user = request.getParameter("user");
        DirContext ctx = new InitialDirContext();
        ctx.search("dc=example", "(uid=" + user + ")", null);
    }
}

File: tests/fixtures/generic/vulnerable/taint_ldap.cs
// @audit-fixture
// @rule: taint_ldap
// @expected: detected
using System.DirectoryServices;
public class AuthController : Controller {
    public IActionResult Login() {
        var user = Request.Form["user"];
        var searcher = new DirectorySearcher();
        searcher.Filter = "(uid=" + user + ")";
        var result = searcher.FindOne();
        return Ok(result != null);
    }
}
